src/gluonnlp/model/bert.py

```diff
@@ -276,11 +276,13 @@ class BERTModel(Block):
         See document of `mx.gluon.Block`.
 
     Inputs:
-        - **inputs**: input sequence tensor of shape (batch_size, seq_length)
-        - **token_types**: input token type tensor of shape (batch_size, seq_length).
+        - **inputs**: input sequence tensor, shape (batch_size, seq_length)
+        - **token_types**: input token type tensor, shape (batch_size, seq_length).
             If the inputs contain two sequences, then the token type of the first
             sequence differs from that of the second one.
-        - **valid_length**: tensor for valid length of shape (batch_size)
+        - **valid_length**: optional tensor of input sequence valid lengths, shape (batch_size)
+        - **positions**: optional tensor of position of tokens for decoding,
+            shape (batch_size, num_positions)
 
     Outputs:
         - **sequence_outputs**: output tensor of sequence encodings.
```

Comment:
Good point. I agree that we need to also provide the option to output a list of sequence encoding from all layers. This can be addressed in a separate PR. Also as suggested in https://github.com/dmlc/gluon-nlp/issues/496  we should provide separate classes for each of these common use cases, otherwise the current BERTModel is too generic and cumbersome.